src/documents/templates/ExtractItem.ts

```diff
@@ -20,19 +21,26 @@ export class ExtractItem {
         let description = await this.ui.showInputBox({ prompt: "Description?", placeHolder: leaveEmpty });
         const insertText = `[parameters('${name}')]`;
         const texts = this.fixExtractTexts(selectedText, insertText, selection, template, editor);
+        let topLevel = this.getTopLevel(template, selection);
         await editor.edit(builder => builder.replace(selection, texts[1]), { undoStopBefore: true, undoStopAfter: false });
-        await new InsertItem(this.ui).insertParameterWithDefaultValue(template, editor, context, name, texts[0], description, { undoStopBefore: false, undoStopAfter: true });
+        await new InsertItem(this.ui).insertParameterWithDefaultValue(topLevel, editor, context, name, texts[0], description, { undoStopBefore: false, undoStopAfter: true });
         editor.revealRange(new vscode.Range(editor.selection.start, editor.selection.end), vscode.TextEditorRevealType.Default);
     }
 
+    private getTopLevel(template: DeploymentTemplateDoc, selection: vscode.Selection): ObjectValue | undefined {
+        let scope = template.getContextFromDocumentLineAndColumnIndexes(selection.start.line, selection.start.character, undefined).getScope();
+        return scope.scopeKind === "NestedDeploymentWithInnerScope" ? scope.rootObject : template.topLevelValue;
+    }
```

Comment:
Looks like you added some code after I pulled down your branch for review...   Will inspect later.  https://github.com/StephenWeatherford/template-examples/blob/master/nested-template-scopes.json has examples of outer and inner-scoped nested templates.